### ISSUE TITLE
use *file* identifier: produces a tree-sitter error

### ISSUE THREAD

**Issue by BigCrow** [nushell/tree-sitter-nu#89]
Using the pattern ``use file identifier`` produces a TS error. This happens when the identifier is not put in a list ex: ``use file [identifier`` - this produces no error. Also happens when * is used instead of an identifier.

Picture (look at the highlighting of the use - on the line before the incorrectly highlighted use there is an TS error, which can be seen via the TSPlayground):
![image](https://github.com/nushell/tree-sitter-nu/assets/1347244/32d83105-a658-416e-92eb-b5f5da1edbd5)

**Comment by fdncred** [comment]
Good catch. Thanks!

**Comment by BigCrow** [comment]
After testing a bit more I think the issue is not actually in the final argument of the use. The actual issue compes from using a pipeline to get the module name.

For example ``use std *`` works properly. However ``use ('std') *`` produces a TS error. However it is valid nu syntax.

**Comment by fdncred** [comment]
This seems like invalid syntax `use ('std') *`. @kubouch Do you know if this syntax is valid?

**Comment by BigCrow** [comment]
The syntax definitely works. The whole point of that syntaxs is to be able to use pipes to provide the module name.

**Comment by BigCrow** [comment]
After looking through the grammar.js, I think this could be a potential fix. On line 121 it is originally:
```js
          field("module", choice($.val_string, $.unquoted)),
```
instead this can be used:
```js
          field("module", choice($.val_string, $.expr_parenthesized, $.unquoted)),
```
From what I saw looking through the grammar expr_parenthesized is used exactly for this type of expression. I tested this via the treesitter cli (both by running test & also by parsing individual files) and it seems to have fixed the issue.

If someone want to give this change a try with their collection of nushell files to see if this creates any ERRORs or incorrect highlighting, give it a go.
